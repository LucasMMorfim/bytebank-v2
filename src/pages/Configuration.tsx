import Card from "../components/Card"
import { Form } from "../components/Form"
import { ParticipantList } from "../components/ParticipantList"
import { Footer } from "../components/Footer"

export const Configuration = () => {
    return (
        <Card>
            <section>
                <h2>Let's start!</h2>
                <Form />
                <ParticipantList />
                <Footer />
            </section>
        </Card>
    )
}