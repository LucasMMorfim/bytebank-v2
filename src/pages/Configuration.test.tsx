import { render } from "@testing-library/react";
import React from "react";
import { RecoilRoot } from "recoil";
import { Configuration } from "./Configuration";

const mockNavigation = jest.fn()

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavigation
    }
})

describe('a pagina de configuracao', () => {
    test('deve ser renderizada corretamente', () => {
        const { container } = render(<RecoilRoot>
            <Configuration />
        </RecoilRoot>)

        expect(container).toMatchSnapshot()
    })
})