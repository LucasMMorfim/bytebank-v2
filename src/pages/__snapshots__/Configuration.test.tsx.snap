// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuracao deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Let's start!
      </h2>
      <form>
        <div
          class="grupo-input-btn"
        >
          <input
            placeholder="Enter the names of the participants"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Add
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="rodape-configuracoes"
      >
        <button
          class="botao"
          disabled=""
        >
          Start Game
        </button>
        <img
          alt="Sacolas de compras"
          src="/images/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
